packages/popover/src/vaadin-popover.js

```diff
@@ -904,6 +819,11 @@ class Popover extends PopoverPositionMixin(
     }
 
     if (this.__hasTrigger('focus')) {
+      // Do not restore focus if closed on focusout to allow focus
+      // to be moved on popover content Tab or target Shift + Tab.
+      if (isKeyboardActive()) {
```

Comment:
Previously, we restored focus on <kbd>Tab</kbd> also and then it was moved programmatically. This is no longer the case as we rely on native keyboard navigation. So we need to prevent restoring focus specifically on <kbd>Tab</kbd> and <kbd>Shift</kbd> + <kbd>Tab</kbd>. In other cases (<kbd>Esc</kbd>, outside click) focus should still be restored to the target with `focus` or `click` trigger.

Reply:
Restored the original logic so this is no longer relevant.